src/main/kotlin/com/coder/gateway/cli/CoderCLIManager.kt

```diff
@@ -199,9 +199,11 @@ class CoderCLIManager(
 
     /**
      * Configure SSH to use this binary.
+     *
+     * This can take a version for testing purposes only.
      */
-    fun configSsh(workspaceNames: List<String>) {
-        writeSSHConfig(modifySSHConfig(readSSHConfig(), workspaceNames))
+    fun configSsh(workspaceNames: List<String>, version: SemVer? = tryVersion()) {
+        writeSSHConfig(modifySSHConfig(readSSHConfig(), workspaceNames, version))
```

Comment:
Suggestion: refactor to `configSsh(options: configSSHOptions)` where options consists of

```
  workspaceNames: List<String>
  disableAutostart: boolean
```

This allows you to then simplify the logic in line 235 to

```
if (opts.disable-autostart) "--disable-autostart" else null
```

and move the version checking logic outside of `modifySSHConfig()`

Reply:
I was thinking of something like this!  But then it felt weird because sometimes we would use the settings (for header command) and sometimes not (for disable autostart).  Maybe a `supports` object that has keys for what features are supported then I have `settings.disableAutostart && supports.disableAutostart`.

Reply:
Gotcha, that would definitely make things more confusing. Not a blocker for me in any case.